experiment_scheduler/task_manager/task_manager_server.py

```diff
@@ -223,18 +222,27 @@ def _get_process_return_code(self, task_id: str):
     def get_task_log(self, request, context):
         """
         Save an output of the requested task and return output file path.
-        If status of the requeest task is Done, delete it from task manager.
+        If status of the request task is Done, delete it from task manager.
+        The response is sent by streaming.
         """
         target_process = self._get_task(request.task_id)
         if target_process is None:
-            return TaskStatus(logfile_path="")
-
-        log_file_path = osp.join(self.log_dir, f"{request.task_id}_log.txt")
+            return TaskLogFile(log_file=None)
 
         if self._get_process_return_code(request.task_id) is not None:
             del self.tasks[request.task_id]
 
-        return TaskLog(logfile_path=log_file_path)
+        log_file_path = osp.join(request.log_file_path, f"{request.task_id}_log.txt")
+        chunk_size = 1024
+        if os.path.exists(log_file_path):
+            with open(log_file_path, mode="rb") as f:
+                while True:
```

Comment:
while True -> while chunk  looks more safer.
Please add try,catch in case of  f.read's failure